f3probe.c

```diff
@@ -507,6 +516,13 @@ static int test_device(struct args *args)
 		report_ops("Reset", reset_count, reset_time_us);
 	}
 
+	if (fake_type == FKTY_CHAIN || fake_type == FKTY_LIMBO || fake_type == FKTY_CHAIN)
+		if (args->fix) {
+			char *cmd = NULL;
+			asprintf(&cmd, "f3fix --last-sec=%" PRIu64 " %s", last_good_sector, final_dev_filename);
+			system(cmd);
+			free(cmd);
```

Comment:
1. We no longer need the two first lines.
2. Move line `free(cmd);` just outside of the `if` statement, so `fix_cmd` is always released; this is important in the case `args->fix` is false, but there is a string at `fix_cmd`.